common/models/transducer/transducer_fullsum.py

```diff
@@ -229,11 +229,15 @@ def make(self, encoder: LayerRef):
     blank_idx = self.ctx.blank_idx
 
     rec_decoder = {
-      "am0": {"class": "gather_nd", "from": _base(encoder), "position": "prev:t"},  # [B,D]
+      "index": {"class": "eval", "from": ["prev:t", "enc_seq_len"], "eval": 'tf.minimum(source(0), source(1)-1)'},
+      "am0": {"class": "gather_nd", "from": _base(encoder), "position": "index"},  # [B,D]
       "am": {"class": "copy", "from": "am0" if search else "data:source"},
 
+      "prev_output_wo_b": {
+        "class": "masked_computation", "unit": {"class": "copy", "initial_output": 0},
+        "from": "prev:output_", "mask": "prev:output_emit", "initial_output": 0},
```

Comment:
I don't understand. Why is this needed? Esp in search, this should have no effect.

Reply:
`prev:output_` doesn't guarantee non_blank during search. Both are sparse, but it messes up the embedding that happens in `slow_rnn`.

I get something like this:
```
TensorFlow exception: indices[0] = 1056 is not in [0, 1056)
         [[node output/rec/slow_rnn/masked/input_embed/linear/embedding_lookup (defined at /Users/mikel/setups/rt4/returnn/returnn/tf/layers/basic.py:1468) ]]

Errors may have originated from an input operation.
Input Source operations connected to node output/rec/slow_rnn/masked/input_embed/linear/embedding_lookup:
 output/rec/slow_rnn/masked/input_embed/linear/py_print_1/Identity (defined at /Users/mikel/setups/rt4/returnn/returnn/tf/util/basic.py:6245)
```

Reply:
Hm, this is strange. Haven't we used it always like this in the other configs as well? Why did the problem never occur? Also, I have used exactly this config, and it did not occur for me. How can that be?

What TensorFlow version do you use?

Also, maybe we should fix `MaskedComputationLayer` instead? This can only happen for frames for `slow_rnn` which will actually not be used (due to the masking). It does not really matter what we calculate in those masked-out frames. We could simply fix the input for the masked-out frames.

But first I want to understand better why this happens now and not before, and not for me.

Reply:
>  Haven't we used it always like this in the other configs as well? 

It looks the same in other configs, I don't get it either.

> What TensorFlow version do you use?

2.4.1

> This can only happen for frames for slow_rnn which will actually not be used (due to the masking)

Exactly.

> But first I want to understand better why this happens now and not before, and not for me.
[Here](https://gist.github.com/jotix16/7a80fa63cc2409f7fe29fc37a739ab3b) are my logs.